### ISSUE TITLE
mgit bootstrap fails randomly due to concurrency

### ISSUE THREAD

**Issue by Reinmar** [cksource/mgit2#23]
https://travis-ci.org/ckeditor/ckeditor5-undo/builds/193346289#L1348:

```
 @ckeditor/ckeditor5-utils                                           20/31 (65%) 
fatal: destination path '/home/travis/build/ckeditor/ckeditor5-undo/packages/ckeditor5-utils' already exists and is not an empty directory.
```

I've seen this error couple of times already.

**Comment by Reinmar** [comment]
OK, the error comes from the fact that we try to install [`@ckeditor/ckeditor5-paragraph`](https://travis-ci.org/ckeditor/ckeditor5-undo/builds/193359811#L1328) after installing [`ckeditor5-paragraph`](https://travis-ci.org/ckeditor/ckeditor5-undo/builds/193359811#L1334).

We need to decide what the name in `mgit.json`'s deps is supposed to be. I think it should mean the package name. Initially, it was the directory name, but now directory is returned by the resolver.

**Comment by Reinmar** [comment]
I'll fix it in https://github.com/ckeditor/ckeditor5-dev so the `mgit.json` is created with proper scoped names. This will prevent conflicts. But we'll still need to fix documentation in mgit2 to match what I described in the previous comment.

**Comment by pomek** [comment]
> OK, the error comes from the fact that we try to install @ckeditor/ckeditor5-paragraph after installing ckeditor5-paragraph.

The error comes because one thread wants to install `ckeditor5-paragraph` and another thread wants to install the same package. The first thread installed it correctly, the second one tries to install it too. After checking the condition "check whether the package exists" (which returns `false`), the package appeared (in the first thread) and the second thread cannot clone it.

I try to show it:

```
first thread - install @ckeditor/ckeditor5-paragraph
second thread - install ckeditor5-paragraph
first thread - resolve directory for the package: packages/ckeditor5-paragraph
second thread - resolve directory - packages/ckeditor5-paragraph
first thread - check whether the directory exists - false
second thread - the same as the first
first thread - clone package
second thread - clone package - throws the error because first thread was faster
```

To be honest, I don't know how we can prevent it.

**Comment by Reinmar** [comment]
> The error comes because one thread wants to install ckeditor5-paragraph and another thread wants to install the same package.

Well, it's only partially true. I explained why we dispatch the bootstrap command for twice for the same repository – it's because of the name. First mgit takes `ckeditor5-paragraph` from `mgit.json` and then gets `@ckeditor/ckeditor5-paragraph` from some dependency `package.json`.

The solution is, as I mentioned, consistent naming. I fixed ckeditor5-dev-tests already (https://github.com/ckeditor/ckeditor5-dev/issues/50) and ckeditor5 (https://github.com/ckeditor/ckeditor5/commit/4e3cc38abfc285abd1cc97271b95c5fce70af5f7).

In mgit itself we just need to fix the documentation to clearly state what is the name supposed to mean.
